k8s/pkg/registryserver/registry_cache.go

```diff
@@ -134,27 +134,43 @@ func (rc *registryCacheImpl) GetEndpointsByNsm(nsmName string) []*v1.NetworkServ
 	return rc.networkServiceEndpointCache.GetByNetworkServiceManager(nsmName)
 }
 
+const maxAllowedAttempts = 10
+
 func (rc *registryCacheImpl) CreateOrUpdateNetworkServiceManager(nsm *v1.NetworkServiceManager) (*v1.NetworkServiceManager, error) {
 	existingNsm := rc.networkServiceManagerCache.Get(nsm.GetName())
 
-	if existingNsm != nil {
-		nsm.ObjectMeta = existingNsm.ObjectMeta
-		logrus.Infof("NSM with name %v already exist in cache, updating: %v", nsm.GetName(), nsm)
-		return rc.updateNetworkServiceManager(nsm)
-	}
-
-	logrus.Infof("Creating NSM: %v", nsm)
-	createNsm, err := rc.addNetworkServiceManager(nsm)
-	if err != nil || apierrors.IsAlreadyExists(err) {
-		existingNsm, err = rc.clientset.NetworkservicemeshV1().NetworkServiceManagers(rc.nsmNamespace).Get(nsm.Name, metav1.GetOptions{})
-		if err != nil {
-			return existingNsm, err
+	attempt := 0
+	for attempt < maxAllowedAttempts {
```

Comment:
Why not `for attempt := 0; attempt < mazAllowedAttempts; attempt ++`?

Reply:
Absolutely agree, I was adding attempts later and accidently do it in such stupid way :) Will fix